artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerSenderContext.java

```diff
@@ -857,10 +561,17 @@ void deliver() {
          // This is discounting some bytes due to Transfer payload
          int frameSize = protonSession.session.getConnection().getTransport().getOutboundFrameSizeLimit() - 50 - (delivery.getTag() != null ? delivery.getTag().length : 0);
 
+         DeliveryAnnotations deliveryAnnotationsToEncode;
+
+         if (reference.getProtocolData() != null && reference.getProtocolData() instanceof DeliveryAnnotations) {
+            deliveryAnnotationsToEncode = (DeliveryAnnotations)reference.getProtocolData();
+         } else {
+            deliveryAnnotationsToEncode = null;
+         }
```

Comment:
This doesnt account for the older explicitly set delivery annotations in the same way the standard message send process does. Seems like they should behave the same?

Reply:
they do, right?

Reply:
this is for large messages...

I was not  adding the deprecated method to this use. (I don't intent to at least).

Reply:
The 'large' message are still AMQPMessage instances. I see no reason the behaviour should differ between them (here or anywhere else really). AMQP itself makes no distinction.